### PR TITLE
Summary has to be converted to HTML / Don't use summary when we have the source

### PR DESCRIPTION
**Pull request by annando**
This should fix the issues that had been mentioned in the PR https://github.com/friendica/friendica/pull/6890

### CODE REVIEW

File: src/Protocol/ActivityPub/Processor.php [hunk 1]
```diff
@@ -138,22 +138,27 @@ public static function updateItem($activity)
 		$item['changed'] = DateTimeFormat::utcNow();
 		$item['edited'] = $activity['updated'];
 		$item['title'] = HTML::toBBCode($activity['name']);
-		$item['content-warning'] = HTML::toBBCode($activity['summary']);
 
-		$content = HTML::toBBCode($activity['content']);
-		$content = self::replaceEmojis($content, $activity['emojis']);
-		$content = self::convertMentions($content);
+		if (!empty($activity['source'])) {
```
Comment (MrPetovan) [inline src/Protocol/ActivityPub/Processor.php]: Why do we skip everything if we have the source?

Comment (annando) [reply src/Protocol/ActivityPub/Processor.php]: The "source" field contains the pure BBCode. This is transmitted in addition to the HTML code that we are transmitting to keep compatibility. Having the source means that we don't have to perform any conversions for Mastodon emojis or other stuff.

And it should fix the issue that an answer to a comment from Friendica to Friendica now always includes a mentioning, when it had been transmitted via AP.

But: The code between processing an "update" and a "create" is so similar, that I guess it is much better to put this similar code in a single function.

**Comment by tobiasd** [comment]
The Summery still has issues

```Myspace lost all the music its users uploaded between 2003 and 2015 · Cory Doctorow – Boing Boing

Do your backups :-D and test to restore them obviously... <a href="https://boingboing.net/2019/03/17/facebook-is-next.html" target="_blank">boingboing.net/2019/03/17/face&hellip;</a> ```

That `a`-tag should not be shown in the output on Mastodon. Question is if it is the fault of Friendica.

**Comment by annando** [comment]
Possibly Mastodon treats the "summary" field as text only.

**Pull request merged**